env.go

```diff
@@ -212,6 +212,26 @@ func set(t reflect.Type, f reflect.Value, value string) error {
 			return err
 		}
 		f.SetUint(v)
+	case reflect.Slice:
+		separator := os.Getenv("ENV_SLICE_SEPARATOR")
```

Comment:
Would it make sense to handle this in the struct tags instead of env var? I wrote it up real quick and it seems to behave the way I would expect: https://github.com/patricksanders/go-env/commit/6a5bf44e0897df0d34f2ee3d8c7d024a5403afaf

Reply:
Yea, i like this. especially if there are reasons out of the users control for why one field might need `|` but another would need `&` or `?`. implemented in [25425e5](https://github.com/Netflix/go-env/pull/13/commits/25425e59b3cfcc3250f04528d943bad6a6523955)